pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -84,6 +91,14 @@ func (d *DispatcherImpl) Shutdown() {
 // Update The Dispatcher's Subscriptions To Align With New State
 func (d *DispatcherImpl) UpdateSubscriptions(subscriberSpecs []eventingduck.SubscriberSpec) map[eventingduck.SubscriberSpec]error {
 
+	// Save the current subscriber specs so that reconfigure() can use them to recreate the Dispatcher if necessary
+	d.currentSubscriberSpecs = subscriberSpecs
+
```

Comment:
Do we want to save them all?  I think instead you want to save the `activeSubscriptions` below on line 158 (after they've been filtered/checked/closed) ?